mycli/packages/special/iocommands.py

```diff
@@ -240,3 +240,54 @@ def execute_system_command(arg, **_):
         return [(None, None, None, response)]
     except OSError as e:
         return [(None, None, None, 'OSError: %s' % e.strerror)]
+
+@export
+def set_redirected_output(append=False, filename=None, enabled=True):
+    global use_redirected_output
+    use_redirected_output = {'append': append, 'filename':filename, 'enabled': enabled}
+
+@export
+def get_redirected_output():
+    return use_redirected_output
+
+@export
+def redirectmatch(sql):
+    if sql[-1] in ("'", '"'): # find filename in quoted redirect
+        quote = sql[-1]
+        m = re.match(r'.*([^\\]'+quote+r'.*?'+quote+'$)', sql)
+        if m:
+            filename = m.group(1)
+        else:
+            filename = None
```

Comment:
Both here and on line 268, you set `filename` to `None`. You can set it to `None` before the conditionals so you don't have to set it twice.